### PR TITLE
Add a soundness check to lint YAML files

### PR DESCRIPTION
**Pull request by ahoppen**

### CODE REVIEW

**Comment by shahmishal** [comment]
Next time, let’s try to separate formatting and feature work in separate PRs. This makes it easier to review the changes.

**Comment by ahoppen** [comment]
https://github.com/swiftlang/github-workflows/pull/15 formats the files and https://github.com/swiftlang/github-workflows/pull/13 now depends on #15.

File: .github/workflows/soundness.yml [hunk 1]
```diff
@@ -51,6 +51,14 @@ on:
         type: string
         description: "Container image for the shell check job. Defaults to latest Swift Ubuntu image."
         default: "swift:6.0-noble"
+      yamllint_enabled:
+        type: boolean
+        description: "Boolean to enable the YAML lint job. Defaults to true."
+        default: true
+      yamllint_container_image:
```
Comment (ahoppen) [inline .github/workflows/soundness.yml:58]: @shahmishal Would you like to have a separate option for the base image of `yamllint`, `spellcheck` and `flake8`? Or should all of this be one single options?

File: .github/workflows/soundness.yml [hunk 2]
```diff
@@ -177,3 +185,21 @@ jobs:
         run: |
           apt-get -qq update && apt-get -qq -y install shellcheck
           git ls-files -z '*.sh' | xargs -0 shellcheck
+
+  yaml-lint:
+    name: Lint YAML
+    if: ${{ inputs.yamllint_enabled }}
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ inputs.yamllint_container_image }}
```
Comment (FranzBusch) [inline .github/workflows/soundness.yml:194]: I don't think we need a container image at all here we can just run on the runner itself

File: .github/workflows/soundness.yml [hunk 3]
```diff
@@ -177,3 +185,21 @@ jobs:
...
+    timeout-minutes: 5
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Run yamllint
+        run: |
+          apt-get -qq update && apt-get -qq -y install yamllint
```
Comment (FranzBusch) [inline .github/workflows/soundness.yml:203]: The runner itself comes with yamlint preinstalled so we don't have to install it here

**Review by FranzBusch — CHANGES_REQUESTED** [review]

**Review by shahmishal — APPROVED** [review]
LGTM!

**Comment by shahmishal** [comment]
@FranzBusch, please feel free to merge it once you’ve reviewed it.

**Review by FranzBusch — APPROVED** [review]

**Pull request merged**